src/app.js

```diff
@@ -82,26 +82,28 @@ if (module.hot) {
   });
 }
 
-render();
-
 // Authenticate and start the authorization process
-const credentials = authenticate();
-store.dispatch(authenticateUser(credentials));
-
-// Install ServiceWorker and AppCache in the end since
-// it's not most important operation and if main code fails,
-// we do not want it installed
-if ('serviceWorker' in navigator && process.env.ENABLE_SERVICEWORKER === '1') {
-  // eslint-disable-next-line global-require
-  const runtime = require('offline-plugin/runtime');
-
-  runtime.install({
-    onUpdateReady: () => {
-      runtime.applyUpdate();
-    },
-  });
-}
+authenticate()
+  .then(credentials => store.dispatch(authenticateUser(credentials)))
+  .finally(() => {
+    render();
 
-if ('serviceWorker' in navigator && process.env.PROXY) {
-  navigator.serviceWorker.register('/sw-proxy.js');
-}
+    // Install ServiceWorker and AppCache in the end since
```

Comment:
Can we extract this service worker instantiation to a separate function?

Reply:
Solved in PR 1308